### PR TITLE
Feature/better errors

### PR DESCRIPTION
**Pull request by jsundra**
Creates errors in the form of:
```
No method was found for: error()
```
or
```
No method was found for: dot(Number, Number)
```
not perfect yet, but so much better than before.

### CODE REVIEW

**Review by thegoldenmule — APPROVED** [review]

File: Jint/Runtime/Interop/MethodInfoFunctionInstance.cs [hunk 1]
```diff
@@ -116,7 +118,18 @@ public JsValue Invoke(MethodInfo[] methodInfos, JsValue thisObject, JsValue[] js
                 }
             }
 
-            throw new JavaScriptException(Engine.TypeError, "No public methods with the specified arguments were found.");
+            var errStr = $"{_methodName}(";
+            if (arguments.Length > 0)
+            {
+                for (var i = 0; i < arguments.Length; i++)
+                {
+                    errStr += $"{arguments[i].Type}, ";
```
Comment (gdamron) [inline Jint/Runtime/Interop/MethodInfoFunctionInstance.cs:126]: is this safe if `null` or `undefined` are passed as arguments?

**Review by gdamron — COMMENTED** [review]
one question

File: Jint/Runtime/Interop/MethodInfoFunctionInstance.cs (hunk 1 above)
Comment (jsundra) [reply Jint/Runtime/Interop/MethodInfoFunctionInstance.cs:126]: Yep! Those aren't the raw arguments, they're `JsValue` instances, so the arg and `Type` will exist, but with a value of `Null` or `Undefined`

Comment (gdamron) [reply Jint/Runtime/Interop/MethodInfoFunctionInstance.cs:126]: 👍

**Review by gdamron — APPROVED** [review]

**Pull request merged**